src/modals/storage-connections/StorageConnectionsTable.js

```diff
@@ -0,0 +1,340 @@
+    row.cells[0].props.value = ''
+  }
+
+  const replaceAttachedCellWithIconCell = (row, connection) => {
+    row.cells[0].props.value = getIsAttachedIcon(connection)
+  }
+
+  const handleTextInputChange = (newValue, evt, rowIndex, cellIndex) => {
+    setRows((oldRows) => {
+      const newRows = Array.from(oldRows)
+      newRows[rowIndex].cells[cellIndex].props.editableValue = newValue
+      return newRows
+    })
+  }
+
+  const createEditableTextCell = (val, isEditable, rules, rowIndex, cellIndex) => {
+    return ({
+      title: (value, rowIndex, cellIndex, props) => (
+        <EditableTextCell
+          value={value}
+          rowIndex={rowIndex}
+          cellIndex={cellIndex}
+          props={props}
+          isDisabled={!isEditable}
+          handleTextInputChange={handleTextInputChange}
+          inputAriaLabel='connection-property-input'
+        />
+      ),
+      props: {
+        value: val,
+        name: `row${rowIndex}cell${cellIndex}`,
+        rules: rules || [],
+      },
+    })
+  }
+
+  const getCellEditableValue = (cell) => {
+    return cell.props.editableValue
+  }
+
+  const createIScsiCells = (connection, rowIndex) => {
+    let cellIndex = 0
+    if (!connection) {
+      return [
+        createEditableTextCell('', false, [], rowIndex, cellIndex++),
+        createEditableTextCell('', true, [requiredRule], rowIndex, cellIndex++),
+        createEditableTextCell('', true, [requiredRule, portRule], rowIndex, cellIndex++),
+        createEditableTextCell('', true, [requiredRule], rowIndex, cellIndex++),
+      ]
+    }
+    const isEditAllowed = canEditConnection(connection, storageDomain)
+    return [
+      createEditableTextCell(getIsAttachedIcon(connection), false, [], rowIndex, cellIndex++),
+      createEditableTextCell(connection.address, isEditAllowed, [requiredRule], rowIndex, cellIndex++),
+      createEditableTextCell(connection.port, isEditAllowed, [requiredRule, portRule], rowIndex, cellIndex++),
+      createEditableTextCell(connection.target, isEditAllowed, [requiredRule], rowIndex, cellIndex++),
+    ]
+  }
+
+  const cellsCreator = {
+    iscsi: createIScsiCells,
+    // add implementations for cells creators here
+  }
+
+  const getIsAttachedIcon = (connection) => {
+    return connection.isAttachedToDomain ? <i className='fa fa-check' /> : ''
+  }
+
+  const createRows = (type, connections) => {
+    const rows = []
+    connections.forEach((connection, index) => {
+      rows.push(createRow(type, connection, true, index + 1))
+    })
+    if (isNewConnectionOpened) {
+      rows.unshift(createRow(type, null, false, 0))
+      rows[0] = applyCellEdits(rows[0], 'edit', false)
+    }
+    return rows
+  }
+
+  const createRow = (type, connection, areActionsEnabled, rowIndex) => {
+    const createCells = cellsCreator[type]
+    return {
+      cells: createCells(connection, rowIndex),
+      connection: connection,
+      isHoverable: true,
+      disableActions: !areActionsEnabled,
```

Comment:
this should depend from `isInMaintance` and from output from `canEditConnection()`